src/sql/workbench/parts/connection/browser/connection.contribution.ts

```diff
@@ -55,6 +61,48 @@ actionRegistry.registerWorkbenchAction(
 	AddServerAction.LABEL
 );
 
+CommandsRegistry.registerCommand('azdata.connect',
+	function (accessor, args) {
+		const capabilitiesServices = accessor.get(ICapabilitiesService);
+		const notificationService = accessor.get(INotificationService);
+		const connectionManagementService = accessor.get(IConnectionManagementService);
+		if (args && args.serverName && args.providerName
+			&& (args.authenticationType === integrated
+				|| args.authenticationType === azureMFA
+				|| (args.userName && args.password))) {
+			const profile: azdata.IConnectionProfile = {
+				serverName: args.serverName,
+				databaseName: args.database,
+				authenticationType: args.authenticationType,
+				providerName: args.providerName,
+				connectionName: '',
+				userName: args.userName,
+				password: args.password,
+				savePassword: true,
+				groupFullName: undefined,
+				saveProfile: true,
+				id: undefined,
+				groupId: undefined,
+				options: {}
+			};
+			const connectionProfile = ConnectionProfile.fromIConnectionProfile(capabilitiesServices, profile);
+
+			connectionManagementService.connectAndSaveProfile(connectionProfile, undefined, {
+				saveTheConnection: true,
+				showDashboard: true,
+				params: undefined,
+				showConnectionDialogOnError: true,
+				showFirewallRuleOnError: true
+			}).then(result => {
+				if (!result.connected) {
+					notificationService.error(localize('connectionFailure', 'Failed to connect to the server, error code: {0}, detail: {1}', result.errorCode, result.errorMessage));
```

Comment:
Should we use the modal connection error dialog instead of this one? If it fails, would the modal dialog be shown by the `connectAndSaveProfile` method anyhow?